### DISCUSSION TITLE
Modal onClose and onOpen event

### DISCUSSION THREAD

**Discussion by fmaddenflx** [trussworks/react-uswds#2515] [Q&A]
Is there an onClose or onOpen event that I can tap into, or is there a way to plug into the toggleModal function in the ModalRef?

**Comment by werdnanoslen** [comment]
Not directly. Aside from using `onClick` on `ModalToggleButton`s or manually checking `modalRef.current?.modalIsOpen`, you could use a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/MutationObserver) to watch the `body` for the `.usa-js-modal--active` class to be [added/removed](https://github.com/trussworks/react-uswds/blob/main/src/components/Modal/Modal.tsx#L84)
